spec/cache_dir_spec.rb

```diff
@@ -0,0 +1,218 @@
+
+      it "must raise an InvalidManifestFile" do
+        expect {
+          subject[:foo]
+        }.to raise_error(Ronin::Wordlists::InvalidManifestFile, /entry :foo is missing a :filename attribute/)
+      end
+    end
+
+    context "if wordlist type is unsupported" do
+      before do
+        subject.instance_variable_set(:@manifest, { foo: { type: "type", url: "url", filename: "filename" } })
+      end
+
+      it "must raise an InvalidManifestFile" do
+        expect {
+          subject[:foo]
+        }.to raise_error(Ronin::Wordlists::InvalidManifestFile, /unsupported wordlist type: "type"/)
+      end
+    end
+
+    context "for git type" do
+      before do
+        subject.instance_variable_set(:@manifest, { foo: { type: :git, url: "url", filename: "filename" } })
+      end
+
+      it "must return new WordlistRepo" do
+        expect(subject[:foo]).to be_kind_of(Ronin::Wordlists::WordlistRepo)
+      end
+    end
+
+    context "for file type" do
+      before do
+        subject.instance_variable_set(:@manifest, { foo: { type: :file, url: "url", filename: "filename" } })
+      end
+
+      it "must return new WordlistFile" do
+        expect(subject[:foo]).to be_kind_of(Ronin::Wordlists::WordlistFile)
+      end
+    end
+  end
+
+  describe "#each" do
+    context "when block is given" do
+      before do
+        subject.instance_variable_set(:@manifest, { foo: { type: :file, url: "url", filename: "filename" } })
+      end
+
+      it "must yields wordlists" do
+        yielded_values = []
+
+        subject.each do |value|
+          yielded_values << value
+        end
+
+        expect(yielded_values.size).to eq(1)
+        expect(yielded_values[0]).to be_kind_of(Ronin::Wordlists::WordlistFile)
+      end
+    end
+
+    context "when block is not given" do
+      it "must return an enumerator" do
+        expect(subject.each).to be_kind_of(Enumerator)
+      end
+    end
+  end
+
+  describe "#list" do
+    it "must list all wordlists" do
+      expect(subject.list).to eq(["example_wordlist.txt"])
+    end
+  end
+
+  describe "#open" do
+    context "if wordlist with given name exists" do
+      it "must open a wordlist file" do
+        expect(subject.open("example_wordlist")).to be_kind_of(Wordlist::File)
+      end
+    end
+
+    context "if wordlist with given name does not exists" do
+      it "must raise an WordlistNotFound" do
+        expect {
+          subject.open("foo")
+        }.to raise_error(Ronin::Wordlists::WordlistNotFound, /wordlist not found: "foo"/)
+      end
+    end
+  end
+
+  describe "#download" do
+    let(:url) { "https://github.com/Escape-Technologies/graphql-wordlist.git" }
+
+    before do
+      subject.instance_variable_set(:@manifest, { foo: { type: :git, url: "https://github.com/Escape-Technologies/graphql-wordlist.git", filename: "test.txt" } })
+    end
+
+    it "must download wordlist" do
+      # expect(subject).to receive(:system).with('git', 'clone', url).and_return(true)
```

Comment:
Remember that `system` will be called on `WordlistRepo`, since ultimately `CacheDir#download` calls `WordlistDir#download`, which then delegates to either `WordlistRepo.download` or `WordlistFile.download`. So you can either do `expect(WordlistRepo).to receive(:system).with(...).and_return(true)` or you could just check that `expect(subject.wordlist_dir).to recieve(:download)` or you could test `expect(WordlistRepo).to receive(:download)`.